File: 0363/363.cpp
class Solution {
public:
    int maxSumOfRectangleNoLargerThanK(TreeNode* root) {
        if(!root) return 0;
        return 1+max(maxSumOfRectangleNoLargerThanK(root->left),maxSumOfRectangleNoLargerThanK(root->right));
    }
};

File: 0363/363.cs
public class Solution {
    public int MaxSumOfRectangleNoLargerThanK(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(MaxSumOfRectangleNoLargerThanK(root.left), MaxSumOfRectangleNoLargerThanK(root.right));
    }
}